service/src/main/java/tds/config/services/impl/ConfigServiceImpl.java

```diff
@@ -36,4 +41,46 @@ public ConfigServiceImpl(ConfigRepository configRepository) {
             .filter(f -> f.getAuditObject().equals(type))
             .findFirst();
     }
+
+    /**
+     * This method is meant to replace CommonDLL._FormatMessage_SP (line 1975)
+     * <p>
+     * Instead of following the existing pattern in the code of getting the _key (int) from tds_coremessageobject for default translation
+     * or getting the _key (varbinary) from client_messagetranslation and then using that to get the actual message text.
+     * The repository takes care of that in a single database query.
+     * <p>
+     * NOTE: This method differs from _FormatMessage_SP in that it does not take in arguments to replace placeholder text in the messages.
+     * The consumer of the endpoint is expected to handle this.
+     */
+    @Override
+    @Cacheable(CacheType.LONG_TERM)
+    public String getSystemMessage(String clientName, String messageKey, String language, String context, String subject, String grade) {
+        String clientDefaultLanguage = null;
+        Optional<ClientLanguage> maybeClientLanguage = configRepository.findClientLanguage(clientName);
```

Comment:
Could we break this into its own method and provide `@Cacheable(CacheType.LONG_TERM)`?  My thinking is you're doing this query each time we get a message with a new messageKey.  If we cache the default call that is one less query we do when looking up a new key (after the first lookup).